src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Resources/Create/ConditionalCreateResourceHandlerTests.cs

```diff
@@ -0,0 +1,177 @@
+        ILogger<ConditionalCreateResourceHandler> logger = Substitute.For<ILogger<ConditionalCreateResourceHandler>>();
+
+        _conditionalCreateHandler = new ConditionalCreateResourceHandler(
+            fhirDataStore,
+            conformanceProvider,
+            resourceWrapperFactory,
+            _searchService,
+            _mediator,
+            resourceIdProvider,
+            _authService,
+            logger);
+
+        // Setup search service to return no matches (for create scenarios)
+        var searchResults = SearchResult.Empty();
+        searchResults.Results = new List<SearchResultEntry>();
+        
+        _searchService.ConditionalSearchAsync(
+            Arg.Any<string>(),
+            Arg.Any<IReadOnlyList<Tuple<string, string>>>(),
+            Arg.Any<CancellationToken>(),
+            Arg.Any<ILogger>())
+            .Returns(searchResults);
+    }
+
+    [Fact]
+    public async Task GivenAConditionalCreateResourceHandler_WhenUserHasSearchAndCreatePermissions_ThenCreateShouldSucceed()
+    {
+        // Arrange
+        _authService
+            .CheckAccess(DataActions.Read | DataActions.Write | DataActions.Search | DataActions.Create, CancellationToken.None)
+            .Returns(DataActions.Search | DataActions.Create);
+
+        var patient = Samples.GetDefaultPatient();
+        var conditionalParameters = new List<Tuple<string, string>> { new("name", "John") };
+        var request = new ConditionalCreateResourceRequest(patient, conditionalParameters, null);
+
+        // Act & Assert - Should not throw UnauthorizedFhirActionException
+        await _conditionalCreateHandler.Handle(request, CancellationToken.None);
+
+        await _mediator
+            .Received()
+            .Send<UpsertResourceResponse>(Arg.Any<CreateResourceRequest>(), Arg.Any<CancellationToken>());
+    }
+
+    [Fact]
+    public async Task GivenAConditionalCreateResourceHandler_WhenUserHasLegacyReadAndWritePermissions_ThenCreateShouldSucceed()
+    {
+        // Arrange
+        _authService
+            .CheckAccess(DataActions.Read | DataActions.Write | DataActions.Search | DataActions.Create, CancellationToken.None)
+            .Returns(DataActions.Read | DataActions.Write);
+
+        var patient = Samples.GetDefaultPatient();
+        var conditionalParameters = new List<Tuple<string, string>> { new("name", "John") };
+        var request = new ConditionalCreateResourceRequest(patient, conditionalParameters, null);
+
+        // Act & Assert - Should not throw UnauthorizedFhirActionException
+        await _conditionalCreateHandler.Handle(request, CancellationToken.None);
+
+        await _mediator
+            .Received()
+            .Send<UpsertResourceResponse>(Arg.Any<CreateResourceRequest>(), Arg.Any<CancellationToken>());
+    }
+
+    [Fact]
+    public async Task GivenAConditionalCreateResourceHandler_WhenUserHasOnlySearchPermission_ThenUnauthorizedExceptionIsThrown()
```

Comment:
Just a suggestion - should we club all the UnauthorizedFhirActionException facts together into a theory?